shell/platform/android/io/flutter/embedding/android/FlutterActivity.java

```diff
@@ -678,18 +682,43 @@ public void unregisterOnBackInvokedCallback() {
   }
 
   private final OnBackInvokedCallback onBackInvokedCallback =
-      Build.VERSION.SDK_INT >= 33
-          ? new OnBackInvokedCallback() {
-            // TODO(garyq): Remove SuppressWarnings annotation. This was added to workaround
-            // a google3 bug where the linter is not properly running against API 33, causing
-            // a failure here. See b/243609613 and https://github.com/flutter/flutter/issues/111295
-            @SuppressWarnings("Override")
-            @Override
-            public void onBackInvoked() {
-              onBackPressed();
-            }
-          }
-          : null;
+      Build.VERSION.SDK_INT < 33 ? null : createOnBackInvokedCallback();
+
+  @VisibleForTesting
+  protected OnBackInvokedCallback getOnBackInvokedCallback() {
+    return onBackInvokedCallback;
+  }
+
+  @NonNull
+  @TargetApi(33)
+  @RequiresApi(33)
+  private OnBackInvokedCallback createOnBackInvokedCallback() {
+    if (Build.VERSION.SDK_INT >= 34) {
```

Comment:
So for API >= 34, popping will be done by the framework via commitBackGesture. For API <= 33, the framework will handle it by listening to popRoute called by onBackPressed as usual.

Could there be any situations where users that are expecting onBackPressed won't get it and won't be listening for commitBackGesture? Like what if someone has a custom Android route transition and they don't know about these new changes?

Reply:
Ok at the route transition level it won't matter because either way it will just result in a pop. But I wonder if this could mess with users at any other level...

Reply:
> But I wonder if this could mess with users at any other level...

If someone directly sets their own method for `SystemChannels.navigation.setMethodCallHandler`, it might cause a problem. However, using the recommended `WidgetsBindingObserver` to register their own handlers shouldn't be a problem, since the `handleCommitBackGesture` method might delegate work to the `handlePopRoute` method if it's not handled by registered observers.